File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = [s[0]]
        i = 1
        while i < len(s):
            if stack and stack[-1] == s[i]:
                stack.pop(-1)
            else:
                stack.append(s[i])
            i += 1
        return ''.join(stack)
            

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string a = "";
        for(int i=0; i<s.length(); i++){
            if(!a.empty() && a.back()==s[i]){
                a.pop_back();
            }else{
                a.push_back(s[i]);
            }
        }
        return a;
    }
};